src/custom/components/Tokens/FiatBalanceCell.tsx

```diff
@@ -0,0 +1,41 @@
+import { Trans } from '@lingui/macro'
+import { Token, CurrencyAmount } from '@uniswap/sdk-core'
+import { useHigherUSDValue } from 'hooks/useUSDCPrice'
+import { FIAT_PRECISION } from 'constants/index'
+import { useActiveWeb3React } from 'hooks/web3'
+import { BalanceValue, BalanceInfoCircle, FiatValue } from './styled'
+import { formatSmart } from 'utils/format'
+import { MouseoverTooltip } from 'components/Tooltip'
+import Loader from 'components/Loader'
+
+type FiatBalanceCellProps = {
+  balance: CurrencyAmount<Token> | undefined
+}
+
+export default function FiatBalanceCell({ balance }: FiatBalanceCellProps) {
+  const { account } = useActiveWeb3React()
+  const hasBalance = balance?.greaterThan(0)
+  const fiatValue = useHigherUSDValue(balance)
+  const formattedFiatValue = formatSmart(fiatValue, FIAT_PRECISION, {
+    thousandSeparator: true,
+    isLocaleAware: true,
+  })
+
+  if (!balance && account) return <Loader />
+  if (!hasBalance) return <BalanceValue hasBalance={false}>0</BalanceValue>
+
+  return (
+    <BalanceValue title={fiatValue?.toExact()} hasBalance={!!hasBalance}>
```

Comment:
Same comment as previous regarding `?.` and title amount formating

Reply:
Update to use formatMax, but here the fiatValue can be null since we have some following logic where we want to display tooltip when there is balance but no fiatBalance.